scripts/add-event.js

```diff
@@ -29,45 +28,40 @@ const price = argv[5];
 const collective_price = argv[6];
 const name = argv[7];
 const description = argv[8];
+const username = argv[9];
+const password = argv[10];
 
-read({ prompt: "Username:" }, function (err, username) {
-  if (err) {
-    console.error("Error obtaining username:", err);
-    process.exit(1);
-  }
-  read({ prompt: "Password:", silent: true }, function (err, password) {
-    if (err) {
-      console.error("Error obtaining password:", err);
-      process.exit(1);
-    }
-    const firebaseConfig = {
-      apiKey,
-      projectId,
+const firebaseConfig = {
+  apiKey,
+  projectId,
+};
+firebase.initializeApp(firebaseConfig);
+firebase
+  .auth()
+  .signInWithEmailAndPassword(username, password)
+  .then(function () {
+    const event = {
+      name,
+      descriptions: [description],
+      start_utc_seconds,
+      duration_minutes,
+      price,
+      collective_price,
     };
-    firebase.initializeApp(firebaseConfig);
     firebase
-      .auth()
-      .signInWithEmailAndPassword(username, password)
+      .firestore()
+      .doc(`venues/${venueId}`)
+      .collection("events")
+      .add(event)
       .then(function () {
-        const event = {
-          name,
-          descriptions: [description],
-          start_utc_seconds,
-          duration_minutes,
-          price,
-          collective_price,
-        };
-        firebase
-          .firestore()
-          .doc(`venues/${venueId}`)
-          .collection("events")
-          .add(event)
-          .catch(function (err) {
-            console.error("Add event error:", err);
-          });
+        process.exit(0);
       })
       .catch(function (err) {
-        console.error("Login error:", err);
+        console.error("Add event error:", err);
+        process.exit(0);
       });
+  })
+  .catch(function (err) {
+    console.error("Login error:", err);
+    process.exit(0);
   });
```

Comment:
We could simplify the nesting here by using `async`/`await` patterns.

Reply:
I did notice some issues there that async/await might resolve. Regarding the username/password, no worries — I was simply sharing this script at the request of @mike-lvov to show how I was reading in the data, but no need to pull it into the main repo, as there aren't really any core changes to the content from the current version.